rest_framework/fields.py

```diff
@@ -234,17 +232,25 @@ def get_error_detail(exc_info):
     with the `code` populated.
     """
     code = getattr(exc_info, 'code', None) or 'invalid'
-
     try:
         error_dict = exc_info.error_dict
     except AttributeError:
+        for error in exc_info.error_list:
+            error.params = error.params or {}
+            error.params[error.code] = error.params.get('limit_value', '')
```

Comment:
I've figured out what the rest of this is doing, but I can't figure out what this line is supposed to be doing.

It's not clear where this parameter was being previously injected, or if it was, and I also don't see anything explaining what this is supposed to be doing.

Reply:
Django's [BaseValidator](https://github.com/django/django/blob/stable/2.1.x/django/core/validators.py#L308-L336) takes a `limit_value` and that value is used for formatting messages. For example, this is the message  template of `MaxValueValidator`.
`message = _('Ensure this value is less than or equal to %(limit_value)s.')`

But in Rest Framework, Serializers use [`code`](https://github.com/django/django/blob/stable/2.1.x/django/core/validators.py#L342) property of Validators as keyword for error message templates. This is IntegerField error message, for example:
`'Ensure this value is less than or equal to {max_value}.`

This block of code extracts needed keyword from it's Validator classes and adds them to params dict for error formatting.

Reply:
So, just to confirm, this is an issue with the current DRF logic and it's just being fixed in this PR alongside the stated removal of `lazy()`?

Reply:
Yes. Actually a performance issue.

Reply:
Given that this is new logic, this is where I lean towards having new tests to cover this new logic. It's clear that there are no current tests that cover it (otherwise they would be failing), and I'd hate to see a regression in the future because they are not covered.

I'd also love to see it broken out into a separate PR, so it can be independently reviewed and not hold up the general refactor that is happening here, but I'm not stuck on that.